docs/user-guide.md

```diff
@@ -0,0 +1,443 @@
+---
+layout: global
+displayTitle: GraphFrames User Guide
+title: User Guide
+description: GraphFrames GRAPHFRAMES_VERSION user guide
+---
+
+This page gives examples of how to use GraphFrames for basic queries, motif finding, and
+general graph algorithms.  This includes code examples in Scala and Python.
+
+* Table of contents
+{:toc}
+
+*Note: Most examples use the GraphFrame from the first subsection:
+[Creating GraphFrames](user-guide.html#creating-graphframes).*
+
+# Creating GraphFrames
+
+Users can create GraphFrames from vertex and edge DataFrames.
+* *Vertex DataFrame*: A vertex DataFrame should contain a special column named "id" which specifies
+  unique IDs for each vertex in the graph.
+* *Edge DataFrame*: An edge DataFrame should contain two special columns: "src" (source vertex ID
+  of edge) and "dst" (destination vertex ID of edge).
+
+Both DataFrames can have arbitrary other columns.  Those columns can represent vertex and edge
+attributes.
+
+A GraphFrame can also be constructed from edge DataFrame, without a vertex DataFrame specified.
```

Comment:
... from a single DataFrame containing edge information. The vertices will be inferred from the source and destinations of the edges.